kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/PaginatedScanSuite.scala

```diff
@@ -0,0 +1,343 @@
+import java.util.Optional
+
+import io.delta.golden.GoldenTableUtils
+import io.delta.kernel.Snapshot
+import io.delta.kernel.data.FilteredColumnarBatch
+import io.delta.kernel.defaults.DeltaTableWriteSuiteBase
+import io.delta.kernel.defaults.engine.{DefaultEngine, DefaultJsonHandler, DefaultParquetHandler}
+import io.delta.kernel.defaults.test.AbstractTableManagerAdapter
+import io.delta.kernel.defaults.utils.{ExpressionTestUtils, TestUtils}
+import io.delta.kernel.internal.PaginatedScanImpl
+import io.delta.kernel.internal.replay.{PageToken, PaginatedScanFilesIteratorImpl}
+
+import org.apache.spark.sql.catalyst.plans.SQLHelper
+import org.scalatest.funsuite.AnyFunSuite
+import org.slf4j.{Logger, LoggerFactory};
+
+class PaginatedScanSuite extends AnyFunSuite with TestUtils
+    with ExpressionTestUtils with SQLHelper with DeltaTableWriteSuiteBase {
+
+  private val logger = LoggerFactory.getLogger(classOf[PaginatedScanSuite])
+
+  private object TestConstants {
+    val DEFAULT_BATCH_SIZE = 5
+  }
+
+  val tableManager: AbstractTableManagerAdapter = getTableManagerAdapter
```

Comment:
`TestUtilsWithTableManagerAPIs` -- could you please use this trait instead of `TestUtils`

https://github.com/delta-io/delta/blob/a7fc9bda3c36430572d7efb22e062c15b26a01ae/kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableReadsSuite.scala#L67C10-L67C39